lib/private/Support/Subscription/Registry.php

```diff
@@ -55,20 +81,29 @@ public function __construct(IConfig $config) {
 	 * @since 17.0.0
 	 */
 	public function register(ISubscription $subscription): void {
-		if ($this->subscription !== null) {
+		if ($this->subscription !== null || $this->subscriptionService !== null) {
 			throw new AlreadyRegisteredException();
 		}
 		$this->subscription = $subscription;
 	}
 
+	public function registerService(string $subscriptionService): void {
+		if ($this->subscriptionService !== null || $this->subscriptionService !== null) {
```

Comment:
```suggestion
		if ($this->subscriptionService !== null) {
```

Reply:
Or 

```suggestion
		if ($this->subscription !== null || $this->subscriptionService !== null) {
```

:man_shrugging: